src/components/TextInput/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Input,
+  Container,
+  ContainerError,
+  TextError,
+} from './styles';
+
+import {Label } from '../Label'
+import { Colors } from '../../utils/Colors';
+
+export const TextInput = ({
+  isPassword = true,
+  color,
+  placeholder,
+  label,
+  active,
+  errorInput = false, 
+  msgError = "",
+  ...rest }) => {
+
+  const [isFocused, setIsFocused] = useState(false);
+  const [border, setBorder] = useState(false);
+
+  useEffect(() => {
+    if(isFocused && active){
+      setBorder(true)
+    }else{
+      if(isFocused && !active){
+        setBorder(true)
+      }else{
+        setBorder(false)
+      }
+    }
```

Comment:
Isso tá muito complexo, analisa como simplificar essa lógica.
```suggestion
    if(isFocused ) {
      setBorder(true)
    } else {
      setBorder(false)
    }
    
```